### DISCUSSION TITLE
Welcome to mobile-react-native Discussions!

### DISCUSSION THREAD

**Discussion by santoshshinde2012** [NeoSOFT-Technologies/mobile-react-native#1] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by sushantsi** [comment]
Hello Team,

when we think about boilerplate it consists of folder structure, store management, middleware, test cases, logging, network interceptors, etc, but now we need to scale up the react-native to the next level step by step, for we have to think about react native web support, Tv platform(there are lots of OTT based application), common app features like Deeplinks, app widget, CICD pipeline. for eg if focus on react native web, then in upcoming years we are developing a single code base for mobile, and web platforms, then it will reduce development time.

for now, we can consider the following things in our boilerplate,

- Architecture
- Common scalable components
- Store management(Redux / Context API) with any middleware
- Common network requests with SSL support
- Security
- CICD
- Crash analytics
- Dark/ light theme support
- Design support for foldable/ flip device
- Code commenting
- Code formatting (prettier) will help when we are working with the team. it will reduce unnecessary conflict
- Husky
- don't use static text, we have to use i18n then in the future if we needed then we can support multilingual
- Hermes - this will help to optimize performance, and reduce app launch time
